ServerCore/Helpers/NotificationHelper.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using ServerCore.DataModel;
+
+/// <summary>
+/// Content of a notification, intended to be broadcast to all players in a specific event, all players on a specific team, or a specific player.
+/// </summary>
+public class Notification
+{
+    public string ID { get; } = Guid.NewGuid().ToString();
+
+    /// <summary>
+    /// The event the notification is for.
+    /// </summary>
+    public int? EventID { get; set; }
+
+    /// <summary>
+    /// The team the notification is for.
+    /// </summary>
+    public int? TeamID { get; set; }
+
+    /// <summary>
+    /// The player the notification is for.
+    /// </summary>
+    public int? PlayerID { get; set; }
+
+    /// <summary>
+    /// The title of the notification.
+    /// </summary>
+    public string Title { get; set; }
+
+    /// <summary>
+    /// The content of the notification.
+    /// </summary>
+    public string Content { get; set; }
+
+    /// <summary>
+    /// The link that should be followed if the notification is clicked.
+    /// </summary>
+    public string LinkUrl { get; set; }
+}
+
+/// <summary>
+/// EventArgs for callers that register for events.
+/// </summary>
+public class NotificationEventArgs : EventArgs
+{
+    /// <summary>
+    /// The notification data
+    /// </summary>
+    public Notification Notification { get; set; }
+}
+
+/// <summary>
+/// Helper object for subscribing to notifications and sending notifications.
+/// </summary>
+public class NotificationHelper
+{
+    /// <summary>
+    /// Singleton for the NotificationHelper object. TODO: Pattern copied from MailHelper; is there any actual reason for this or can it all just be static?
+    /// </summary>
+    private static NotificationHelper Singleton { get; set; }
+
+    private bool IsDev;
+    private Dictionary<int, List<NotificationListener>> listenersByEvent = new Dictionary<int, List<NotificationListener>>();
```

Comment:
How about the List then? Does that also need to be threadsafe, and if so which primitive makes sense? I need to add, remove, and enumerate, and none of the concurrent primitives seem to make sense. The closest seems to be another ConcurrentDictionary, which seems wasteful since I don't need values.